shapeshift/web
# fix: trade ratio & CoW quote display

## Description

- Fixes the trade ratio calculation, which was incorrectly handling sell asset fees (which currently only affects CoW Swap quotes)
- Fixes the display of sell asset fees, which were not correctly shown to the user (again, currently only affects CoW Swap quotes)
- Renames `sellAmountBeforeFeesCryptoBaseUnit` to `sellAmountIncludingProtocolFeesCryptoBaseUnit`, because that's what it actually is, and probably caused this bug

This PR is also important as it fixes the ratio data send to MixPanel.

## Pull Request Type


## Issue (if applicable)

N/A

## Risk

Medium - quotes and their ratio/percent better/worse than display might be broken.

## Testing

Get a quote for an asset pair with many supported swappers (e.g. USDC to FOX).
- Ensure that the percent difference looks sane
- Ensure that for CoW Swap quotes, we are subtracting the fee from the receive amount

### Engineering

Inspect CoW quote network requests and ensure that the data matches what we are showing, particularly with regard to fees and receive amount.

### Operations

Check our CoW quotes against https://swap.cow.fi/#/1/swap and ensure they are close.

## Screenshots (if applicable)

<img width="459" alt="Screenshot 2023-08-11 at 4 17 04 pm" src="https://github.com/shapeshift/web/assets/97164662/1525ae8d-8e7e-4015-bacf-edf0d9e539c4">
